src/FlowtideDotNet.Substrait/Sql/Internal/SqlSubstraitVisitor.cs

```diff
@@ -26,13 +29,37 @@ internal class SqlSubstraitVisitor : SqlBaseVisitor<RelationData?, object?>
         private readonly SqlPlanBuilder sqlPlanBuilder;
         private readonly SqlFunctionRegister sqlFunctionRegister;
         private readonly Dictionary<string, CTEContainer> cteContainers;
+        private readonly Dictionary<string, ExchangeContainer> exchangeRelations;
+        private readonly Dictionary<string, ViewContainer> viewRelations;
+        private string? subStreamName;
+        private int exchangeTargetIdCounter;
+        private readonly List<Relation> subRelations;
 
         public SqlSubstraitVisitor(SqlPlanBuilder sqlPlanBuilder, SqlFunctionRegister sqlFunctionRegister)
         {
             this.sqlPlanBuilder = sqlPlanBuilder;
             this.sqlFunctionRegister = sqlFunctionRegister;
             tablesMetadata = sqlPlanBuilder._tablesMetadata;
             cteContainers = new Dictionary<string, CTEContainer>(StringComparer.OrdinalIgnoreCase);
+            exchangeRelations = new Dictionary<string, ExchangeContainer>(StringComparer.OrdinalIgnoreCase);
+            viewRelations = new Dictionary<string, ViewContainer>(StringComparer.OrdinalIgnoreCase);
+            subRelations = new List<Relation>();
+        }
+
+        public List<Relation> GetRelations(Sequence<Statement> statements)
```

Comment:
```suggestion
        public IEnumerable<Relation> GetRelations(Sequence<Statement> statements)
```
Is there any benefit for returning the list directly here? Could it be immutable or ienumerable?

Reply:
It has to be a list right now since that is what is added into the plan, this might change if the entire plan is made immutable.